django/db/models/deletion.py

```diff
@@ -308,7 +308,7 @@ def collect(self, objs, source=None, nullable=False, collect_related=True,
                     model.__name__,
                     ', '.join(protected_objects),
                 ),
-                chain.from_iterable(protected_objects.values()),
+                list(chain.from_iterable(protected_objects.values())),
```

Comment:
We can revert this change and fix collecting protected objects (see [comment](https://github.com/django/django/pull/12394/files#r373228132)).